ecoscope/io/async_earthranger.py

```diff
@@ -0,0 +1,719 @@
+                    continue
+
+                observations_by_subject["subject_id"] = subject_id
+                observations_by_subject["created_at"] = pd.to_datetime(
+                    observations_by_subject["created_at"],
+                    errors="coerce",
+                    utc=True,
+                ).dt.tz_convert(tz)
+
+                observations_by_subject["recorded_at"] = pd.to_datetime(
+                    observations_by_subject["recorded_at"],
+                    errors="coerce",
+                    utc=True,
+                ).dt.tz_convert(tz)
+                observations_by_subject.sort_values("recorded_at", inplace=True)
+                observations_by_subject = AsyncEarthRangerIO._to_gdf(observations_by_subject)
+
+                if relocations:
+                    observations_by_subject = ecoscope.base.Relocations.from_gdf(
+                        observations_by_subject,
+                        groupby_col="subject_id",
+                        uuid_col="id",
+                        time_col="recorded_at",
+                    )
+
+                if patrol_types is not None:
+                    patrol_type = patrol_types[
+                        patrol_types["value"] == patrol_segment.get("patrol_type")
+                    ].reset_index()["id"][0]
+                    observations_by_subject["patrol_id"] = patrol["id"]
+                    observations_by_subject["patrol_serial_number"] = patrol["serial_number"]
+                    observations_by_subject["patrol_start_time"] = patrol_start_time
+                    observations_by_subject["patrol_end_time"] = patrol_end_time
+                    observations_by_subject["patrol_type"] = patrol_type
+                    observations_by_subject = (
+                        observations_by_subject.reset_index()
+                        .merge(
+                            patrol_types.add_prefix("patrol_type__"),
+                            left_on="patrol_type",
+                            right_on="id",
+                        )
+                        .drop(
+                            columns=[
+                                "patrol_type__ordernum",
+                                "patrol_type__icon_id",
+                                "patrol_type__default_priority",
+                                "patrol_type__is_active",
+                            ]
+                        )
+                    )
+                    observations_by_subject.set_index("id", inplace=True)
+
+                if len(observations_by_subject) > 0:
+                    observations = pd.concat([observations, observations_by_subject])
+
+            except Exception as e:
+                print(
+                    f"Getting observations for subject_id={subject_id} start_time={patrol_start_time}"
+                    f"end_time={patrol_end_time} failed for: {e}"
+                )
+        return observations
```

Comment:
this part is also very similar with EarthRangerIO client

Reply:
See my comment here: https://github.com/wildlife-dynamics/ecoscope/pull/198/files#r1689730605

It's sort of by-design here because while the logic is the same it has to happen in slightly different places in async land vs sync land. Keen for your thoughts on the solution in the previous comment.